experimental/adaptive-card-prompt-sample/javascript_nodejs/adaptiveCardPrompt.js

```diff
@@ -0,0 +1,222 @@
+        this.validator = validator;
+        this._inputFailMessage = settings.inputFailMessage || 'Please fill out the Adaptive Card';
+        this._requiredInputIds = settings.requiredInputIds || [];
+        this._missingRequiredInputsMessage = settings.missingRequiredInputsMessage || 'The following inputs are required';
+        this._attemptsBeforeCardRedisplayed = settings.attemptsBeforeCardRedisplayed || 3;
+        this._card = settings.card;
+        if (settings.promptId) {
+            this._promptId = settings.promptId;
+            this.usesCustomPromptId = true;
+        }
+    }
+    get inputFailMessage() {
+        return this._inputFailMessage;
+    }
+    set inputFailMessage(message) {
+        this._inputFailMessage = message;
+    }
+    get requiredInputIds() {
+        return this._requiredInputIds;
+    }
+    set requiredInputIds(ids) {
+        this._requiredInputIds = ids;
+    }
+    get missingRequiredInputsMessage() {
+        return this._missingRequiredInputsMessage;
+    }
+    set missingRequiredInputsMessage(message) {
+        this._missingRequiredInputsMessage = message;
+    }
+    get attemptsBeforeCardRedisplayed() {
+        return this._attemptsBeforeCardRedisplayed;
+    }
+    set attemptsBeforeCardRedisplayed(attempts) {
+        this._attemptsBeforeCardRedisplayed = attempts;
+    }
+    get promptId() {
+        return this._promptId;
+    }
+    set promptId(id) {
+        this.usesCustomPromptId = true;
+        this._promptId = id;
+    }
+    get card() {
+        return this._card;
+    }
+    set card(card) {
+        this._card = card;
+    }
+    async beginDialog(dc, options) {
+        // Initialize prompt state
+        const state = dc.activeDialog.state;
+        state.options = options;
+        state.state = {};
+        // Send initial prompt
+        await this.onPrompt(dc.context, state.state, state.options, false);
+        return Dialog.EndOfTurn;
+    }
+    async onPrompt(context, state, options, isRetry) {
+        // Should use GUID for C# -- it isn't native to Node, so this keeps dependencies down
+        // Only the most recently-prompted card submission is accepted
+        this._promptId = this.usesCustomPromptId ? this._promptId : `${Math.random()}`;
+        let prompt = isRetry && options.retryPrompt ? options.retryPrompt : options.prompt;
+        // Create a prompt if user didn't pass it in through PromptOptions
+        if (!prompt || Object.keys(prompt).length === 0 || typeof (prompt) != 'object' || !prompt.attachments || prompt.attachments.length === 0) {
+            prompt = {
+                attachments: [],
+                text: typeof (prompt) === 'string' ? prompt : undefined,
+            };
+        }
+        // Use card passed in PromptOptions or if it doesn't exist, use the one passed in from the constructor
+        const card = prompt.attachments && prompt.attachments[0] ? prompt.attachments[0] : this._card;
```

Comment:
I'd say they're more tightly-coupled. I like the idea of renaming to something like `AdaptiveCardInputPrompt` (since a "form" sounds like multi-input but "input" could be single or multi).

I think I'll require that the card be passed in via constructor so the user can have whatever attachments they want. Good call.